client/src/main/java/com/paypal/selion/configuration/ListenerInfo.java

```diff
@@ -33,24 +38,62 @@
      *            - The JVM argument that should be read to decide if the listener is to be enabled/disabled.
      */
     public ListenerInfo(Class<?> className, String jvmArgToParse) {
+        this(className, jvmArgToParse, ENABLE_LISTENER_BY_DEFAULT);
+    }
+
+    /**
+     * Saves information about listener class which is enabled/disabled per its VM argument. When VM argument is
+     * unavailable, it's enabled/disabled is decided by the caller via (@code isEnabledWhenJvmAbsent}.
+     * 
+     * @param className
+     *            - The Class name.
+     * @param jvmArgToParse
+     *            - The JVM argument that should be read to decide if the listener is to be enabled/disabled.
+     * @param isListenerEnabledWhenJvmAbsent
```

Comment:
How can the JVM be absent in a java process? I suspect you mean to call this more like `defaultStateWhenNotDefined`